lib/ansible/plugins/strategy/__init__.py

```diff
@@ -912,23 +912,8 @@ def _do_handler_run(self, handler, handler_name, iterator, play_context, notifie
             for included_file in included_files:
                 try:
                     new_blocks = self._load_included_file(included_file, iterator=iterator, is_handler=True)
-                    # for every task in each block brought in by the include, add the list
-                    # of hosts which included the file to the notified_handlers dict
-                    for block in new_blocks:
-                        iterator._play.handlers.append(block)
-                        for task in block.block:
```

Comment:
Just as a note, getting rid of `block.block` also means that, if specified, all sections `block: rescue: always:` will be executed instead of being ignored.